packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -27,9 +35,18 @@ export type ReactApplicationExtensionConfig = ApplicationExtensionConfig
 export class ApplicationExtension<
     Config extends ReactApplicationExtensionConfig
 > extends ApplicationExtensionBase<Config> {
+    protected _cachedRoutes: RouteProps[] | null = null
+
     constructor(config: Config) {
         super(config)
         this.extendRoutes = this.extendRoutes.bind(this)
+
+        if (this.getRoutesAsync) {
+            // Deserialize the routes on the client to ensure the latest routes are loaded on the client
```

Comment:
Since this block of code is meant for client side only:
```suggestion
        if (this.getRoutesAsync && !isServerSide) {
            // Deserialize the routes on the client to ensure the latest routes are loaded on the client
```

Reply:
Actually `deserialize()` happens only on client-side while the `applyCacheForMethod` should apply on server side.

This is because we want the result from `getRoutesAsync` to cache when we call it before we serialize on the server.